addon/components/fl-input.ts

```diff
@@ -50,6 +52,11 @@ export default class FlInput<T extends FlInputArgs> extends Component<T> {
         return this.args.id ?? `fl-input-${guidFor(this)}`;
     }
 
+    get name() {
+        const name = this.args.name ?? this.args.placeholder;
+        return name ? dasherize(name.replace(/\W/g, '')) : '';
```

Comment:
looks good i just wouldn't muck with a manually passed in `@name` arg value, i'd do it like:

```ts
get name() {
    return this.args.name ?? dasherize(this.args.placeholder?.replace(/\W/g, '') ?? '');
}
```